core/src/impl/Kokkos_Memory_Fence.hpp

```diff
@@ -57,6 +57,9 @@ void memory_fence() {
 #pragma omp flush
 #elif defined(__HIP_DEVICE_COMPILE__)
   __threadfence();
+#elif defined(__SYCL_DEVICE_ONLY__)
+  sycl::ONEAPI::atomic_fence(sycl::ONEAPI::memory_order::acq_rel,
+                             sycl::ONEAPI::memory_scope::device);
```

Comment:
Do we need/want `memory_order::seq_cst` or `memory_scope::system` here?

Reply:
I think what you did is right